### PR TITLE
fix: [CI-7379]: Add pprof profilers to analyse log service in case of…

### PR DESCRIPTION
**Pull request by smjt-h**
… load test

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45114)

### CODE REVIEW

File: product/log-service/handler/handler.go [hunk 1]
```diff
@@ -38,32 +39,32 @@ func Handler(stream stream.Stream, store store.Store, config config.Config, ngCl
...
-	//	sr.HandleFunc("/debug/pprof/block", pprof.Index)
-	//	sr.HandleFunc("/debug/pprof/goroutine", pprof.Index)
-	//	sr.HandleFunc("/debug/pprof/threadcreate", pprof.Index)
-	//
-	//	return sr
-	//}())
+	// Log service info endpoints
+	// Only accessible from Harness side (admin privileges)
+	// Format: /info
+	r.Mount("/info", func() http.Handler {
```
Comment (vistaarjuneja) [inline product/log-service/handler/handler.go:45]: can you add an environment variable and enable this only if that variable is enabled?
Something like `LOG_SERVICE_DEBUG_MODE`
We'll enable this mode only for non-prod environments

Comment (smjt-h) [reply product/log-service/handler/handler.go]: Added this

File: product/log-service/handler/handler.go [hunk 2]
```diff
@@ -38,32 +40,34 @@ func Handler(stream stream.Stream, store store.Store, config config.Config, ngCl
...
-	//	sr.HandleFunc("/debug/pprof/profile", pprof.Profile)
-	//	sr.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
-	//	sr.HandleFunc("/debug/pprof/trace", pprof.Trace)
-	//	sr.HandleFunc("/debug/pprof/block", pprof.Index)
-	//	sr.HandleFunc("/debug/pprof/goroutine", pprof.Index)
-	//	sr.HandleFunc("/debug/pprof/threadcreate", pprof.Index)
-	//
-	//	return sr
-	//}())
+	if debug := os.Getenv("LOG_SERVICE_DEBUG_MODE"); debug != "" {
```
Comment (vistaarjuneja) [inline product/log-service/handler/handler.go:43]: don't add this here - this should come from the config.
All our config for log service is present in the config variable

Comment (smjt-h) [reply product/log-service/handler/handler.go]: added in config

File: product/log-service/config/config.go [hunk 3]
```diff
@@ -11,7 +11,7 @@ import (
 
 // Config provides the system configuration.
 type Config struct {
-	Debug bool `envconfig:"LOG_SERVICE_DEBUG"`
```
Comment (vistaarjuneja) [inline product/log-service/config/config.go:14]: can you confirm this variable is not being used anywhere?

Comment (smjt-h) [reply product/log-service/config/config.go:14]: yes its not used anywhere i checked

**Review by vistaarjuneja — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45114#-:-NQUrsXnEbrtfwxfj87C:bnfp4nl)* status: :shipit: complete! ✓

**Comment by vistaarjuneja** [comment]
LGTM

**Pull request merged**